marklogic-data-hub-central/ui/src/components/queries/saving/edit-save-query/save-changes-modal.tsx

```diff
@@ -88,18 +94,28 @@ const SaveChangesModal: React.FC<Props> = (props) => {
                 //const response = await updateQuery(currentQuery);
                 const response = await axios.put(`/api/entitySearch/savedQueries`, currentQuery);
                 if (response.data) {
-                    props.setSaveChangesModalVisibility()
+                    setAllSearchFacets(facets);
+                    props.setSaveChangesModalVisibility();
+                    applySaveQuery(searchOptions.query, searchOptions.entityTypeIds, facets, queryName);
+                    props.setCurrentQueryDescription(queryDescription);
                 }
             } catch (error) {
-                handleError(error);
+                if (error.response.status === 400) {
+                    if (error.response.data.hasOwnProperty('message')) {
+                        setErrorMessage(error['response']['data']['message']);
```

Comment:
This seems to be repeated - it's in 3 different places. 

@wooldridge @xnikhil08  It seems there should be a way to reduce this duplication so that each "catch" block doesn't have to do so much work? The problem with repeating this work in multiple places is that if we ever want to change how we handle e.g. 400 errors, we have N places to modify as opposed to just one.

Reply:
@rjrudin the values that are set in the catch are specific to the component. Apart from setErrorMessage state, there are other states that are set in different catch blocks. This error handling is specific to the child components of the Query component. There is also a handleError generic error handler, which handles 400 error.

Reply:
Yeah, I'm not really sure of better alternatives. async await in a try catch block is already a pretty lean. One refactor I would recommend is combining the edit and save modals into one since they share inputs.